lib/utils/useMedia.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+
+import { useCallback, useLayoutEffect, useMemo, useState } from 'react';
+
+import { useOverdriveContext } from '../components/OverdriveProvider';
+import { useTheme } from '../components/ThemeProvider';
+import { Tokens } from '../themes/tokens';
+
+export const useMedia = (
+	queries: ReadonlyArray<keyof Tokens['breakpoints']>,
+	fallbackCase = false,
+): readonly boolean[] => {
+	const theme = useTheme();
+	const { isServer } = useOverdriveContext();
+
+	if (isServer) return queries.map(() => fallbackCase);
+
+	const getQueries = useCallback(
+		() =>
+			queries.map(media => {
+				const [min, max] = theme.breakpoints[media];
+				return makeQueryString(min, max);
+			}),
+		[theme],
+	);
+	const matchesInit = useMemo(
+		() => getQueries().map(query => window.matchMedia(query).matches),
+		[getQueries],
+	);
+	const [matches, setMatches] = useState<readonly boolean[]>(matchesInit);
+
+	useLayoutEffect(() => {
+		let isMounted = true;
+
+		const matchers = getQueries().map(query => window.matchMedia(query));
+
+		// Add listeners
+		const handlers = matchers.map((matcher, idx) => {
```

Comment:
Reading this at first it looked like this is a list of handlers whereas its a list of handler clearers. Not a big deal but might be good to rename to something else e.g. listenerRemovers or something better like that.

Reply:
Yeah makes sense!